include/zephyr/dt-bindings/gpio/nordic-nrf-gpio.h

```diff
@@ -1,74 +1,64 @@
 /*
- * Copyright (c) 2022 Vestas Wind Systems A/S
+ * Copyright (c) 2022 Nordic Semiconductor ASA
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 #ifndef ZEPHYR_INCLUDE_DT_BINDINGS_GPIO_NORDIC_NRF_GPIO_H_
 #define ZEPHYR_INCLUDE_DT_BINDINGS_GPIO_NORDIC_NRF_GPIO_H_
 
 /**
- * @name GPIO drive strength flags
+ * @brief nRF-specific GPIO Flags
+ * @defgroup gpio_interface_nrf nRF-specific GPIO Flags
+ * @ingroup gpio_interface
+ * @{
+ */
+
+/**
+ * @name nRF Drive Flags
  *
- * The drive strength flags are a Zephyr specific extension of the standard GPIO
- * flags specified by the Linux GPIO binding. Only applicable for Nordic
- * Semiconductor nRF SoCs.
+ * Standard (S) or High (H) drive modes can be applied to both pin levels, 0 or
+ * 1. High drive mode will increase current capabilities of the pin (refer to
+ * each SoC reference manual).
  *
- * The drive strength of individual pins can be configured
- * independently for when the pin output is low and high.
+ * When the pin is configured to operate in open-drain mode (wired-and), the
+ * drive mode can only be selected for the 0 level (1 is disconnected).
+ * Similarly, when the pin is configured to operate in open-source mode
+ * (wired-or), the drive mode can only be set for the 1 level
+ * (0 is disconnected).
  *
- * The `GPIO_DS_*_LOW` enumerations define the drive strength of a pin
- * when output is low.
-
- * The `GPIO_DS_*_HIGH` enumerations define the drive strength of a pin
- * when output is high.
+ * The drive flags are encoded in the 8 upper bits of @ref gpio_dt_flags_t as
+ * follows:
  *
- * The interface supports two different drive strengths:
- * `DFLT` - The lowest drive strength supported by the HW
- * `ALT` - The highest drive strength supported by the HW
+ * - Bit 8: Drive mode for '0' (0=Standard, 1=High)
+ * - Bit 9: Drive mode for '1' (0=Standard, 1=High)
  *
  * @{
  */
-/** @cond INTERNAL_HIDDEN */
-#define NRF_GPIO_DS_LOW_POS 9
-#define NRF_GPIO_DS_LOW_MASK (0x1U << NRF_GPIO_DS_LOW_POS)
```

Comment:
Shouldn't all the existing flags be deprecated?

Reply:
the generic ones are already deprecated, these are too recent (and not much used) to bother with deprecation IMO.

Reply:
The SoC specific flags are public API and they were introduced in v3.1.0. Users just had to go through changing their code once, with proper deprecation warnings. Why would we skip it now?

Reply:
They were part of a stable API before, so it made sense to deprecate. The newly created ones were a 1:1 replacement  with vendor prefix, but not necessarily what vendors wanted now that it is possible to create custom flags. Unfortunately, we did not have time to adjust them, and since it's a feature rarely used (there are no uses in zephyr/sdk-nrf), deprecation of an _intermediate_ state doesn't sound like it is necessary.